modules/webgl/src/classes/texture.js

```diff
@@ -340,7 +339,18 @@ export default class Texture extends Resource {
           }
           break;
         case 'compressed':
-          gl.compressedTexImage2D(target, level, format, width, height, border, data);
+          for (const [levelIndex, levelData] of data.entries()) {
```

Comment:
Are we just covering the case where data is an array? Maybe an index for loop would look simpler than `data.entries()`

```
for (const level = 0; level < data.length; level++) {
   const levelData = data[];
}
```

Generally I'd like to change the logic to support mipmap arrays for all types (not just compressed textures) but since that is a bigger change, we can do that after the imminent 8.4 release.